hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/SelectorOptimizer.java

```diff
@@ -163,6 +163,11 @@ public boolean remove(Object o) {
 
         @Override
         public boolean contains(Object o) {
+            for (int i = 0; i < size(); i++) {
```

Comment:
Can you make use of the size field directly instead of the size() method.

Reply:
https://github.com/hazelcast/hazelcast/pull/24267/commits/22bf636e527c2c678d7a7b891279d48734d087d9